agent/engine/task_manager_linux.go

```diff
@@ -15,9 +15,19 @@
 
 package engine
 
+import (
+	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
+	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+// control is used to manipulate cgroups and ease testing
+var control cgroup.Control
```

Comment:
Is there anyway to get rid of the global variable, if it is specific for each task, can you add it to the `mtask` struct?

Reply:
`mtask` is not platform specific and cgroups don't apply to windows. 
To get rid of this global would require some sort of encapsulation and since we only have one resource here, I'm not sure if its necessary.